server/store.go

```diff
@@ -64,6 +64,10 @@ func (p *Plugin) fetchOAuthUserInfo(tokenKey, userID string) (*zoom.OAuthUserInf
 
 func (p *Plugin) disconnectOAuthUser(userID string) error {
 	encoded, err := p.API.KVGet(zoomUserByMMID + userID)
+	if len(encoded) == 0 {
+		return errors.Wrap(err, "Your Zoom account is not currently linked")
+	}
+
 	if err != nil {
 		return errors.Wrap(err, "could not find OAuth user info")
 	}
```

Comment:
Could we actually put the `len(encoded) == 0` block after the error check? And then we would use `errors.New` instead of `errors.Wrap` in that block.

Reply:
According to me yes it will be fine. But since the check happens first it will directly alert the user regarding the unlinked zoom account.

Reply:
So if their account is not linked, will `err` be a valid error or be nil? If it would be nil in that case, then let's have the `len(encoded) == 0` check be afterwards.

Either way, we do not want to use the `err` variable in the `len(encoded) == 0` block. So instead of

`return errors.Wrap(err, "Your Zoom account is not currently linked")`

we want to use

`return errors.New("your Zoom account is not currently linked")`.

@hanzei We are using an `error` to carry a user error message to the UI here, which is why I questioned the capital letter thing here https://github.com/mattermost/mattermost-plugin-zoom/pull/180#issuecomment-713227345 . Do you think we should change how we are processing the user message here to not use `error`'s, or deal with the issue of giving the user an error message sentence that starts with a lower case letter?

Reply:
> So if their account is not linked, will `err` be a valid error or be nil? If it would be nil in that case, then let's have the `len(encoded) == 0` check be afterwards.
> 
> Either way, we do not want to use the `err` variable in the `len(encoded) == 0` block. So instead of
> 
> `return errors.Wrap(err, "Your Zoom account is not currently linked")`
> 
> we want to use
> 
> `return errors.New("your Zoom account is not currently linked")`.

@mickmister So is it fine if I add the  `len(encoded)` check after the `err` check ?

Reply:
@yash2189 I'm not sure, do you know if `err` will be nil when they are unauthenticated? If that's the case, the `len(encoded)` block should go after the `err` block. Otherwise, we don't need the `len(encoded)` block.

Reply:
@mickmister could you please tell me what do I do next ? I'm confused

Reply:
@yash2189 Here's what I think should be done:

1. Use the feature, and analyze what happens in the code when the user is not authenticated.
2. Verify by running the code whether `err` is nil or not when the user is not authenticated.

3a. If your findings determine that `err` is nil in this case then:
- add the `len(encoded)` block
- use `errors.New` in that block

3b. If your findings determine that `err` is not nil in this case then:
- we just need to change the error string being used in the already existing `err != nil` block
- and we do not need to add a new block

I'm guessing that 3a. will be the case, but if I were to author this PR, I would make sure the code does what I think it is supposed to do. This would require me to take the steps I laid out above.